### PR TITLE
feat(ui) change layout of web app based on language

### PR DESCRIPTION
**Pull request by c0m1t**
This is the stepping stone for RTL layout support. It is a commit cherry picked from #11079 pull request.

Layout direction will change to RTL if user chooses Arabic or Persian as the language 

Note that there is a `break` statement in [react/features/base/ui/middleware.web.js](https://github.com/jitsi/jitsi-meet/compare/master...c0m1t:layout-direction?expand=1#diff-34164bdca1f7a5abee4492695758f773f1413b55cb62e24997ff66c4b7b4c211).
```js
    case I18NEXT_INITIALIZED:
    case LANGUAGE_CHANGED :{

        // XXX Right now, removing the following line will mess up with the app layout.
        // Remove the following line only for testing purposes.
        break;
```
If you want to see how app would look like in RTL layout, you should comment out the `break` statement. Note that **changing the direction to RTL in this PR will break the app layout**. To see how app would look like in RTL layout when styles are converted to JSS, see #11079 . Some styles must be converted to JSS(#11079) to prevent messing up the RTL layout. So do not remove the `break` statement while merging this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,6 +77,7 @@
     "jquery": "3.5.1",
     "jquery-i18next": "1.2.1",
     "js-md5": "0.6.1",
+    "jss-rtl": "^0.3.0",
```
Comment (saghul) [inline package.json:80]: Please pin the dependency to exact 0.3.0

File: react/features/base/i18n/i18next.js [hunk 2]
```diff
@@ -106,7 +106,7 @@ i18next.addResourceBundle(
 require('./BuiltinLanguages');
 
 // Label change through dynamic branding is available only for web
-if (typeof APP !== 'undefined') {
+if (navigator.product !== 'ReactNative') {
```
Comment (saghul) [inline react/features/base/i18n/i18next.js:109]: Why this change?

File: react/features/base/ui/actionTypes.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export const SET_DIRECTION = 'SET_DIRECTION';
```
Comment (saghul) [inline react/features/base/ui/actionTypes.js:1]: Please document the action type.

Comment (saghul) [reply react/features/base/ui/actionTypes.js:1]: Also let's call it something like SET_LANGUAGE_DIRECTION

File: react/features/base/ui/components/JitsiThemeProvider.web.js [hunk 4]
```diff
@@ -38,9 +45,10 @@ function JitsiThemeProvider(props: Props) {
  */
 function _mapStateToProps(state) {
     const { muiBrandedTheme } = state['features/dynamic-branding'];
+    const { direction } = state['features/base/ui'];
 
     return {
-        _theme: muiBrandedTheme || BaseTheme
+        _theme: muiBrandedTheme || direction === 'rtl' ? rltTheme : ltrTheme
```
Comment (saghul) [inline react/features/base/ui/components/JitsiThemeProvider.web.js:51]: Doesn't this mean the direction won't work when dynamic branding is used? Maybe we could inject it, if missing?

File: react/features/base/ui/middleware.web.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case I18NEXT_INITIALIZED:
+    case LANGUAGE_CHANGED :{
+
+        // XXX Right now, removing the following line will mess up with the app layout.
```
Comment (saghul) [inline react/features/base/ui/middleware.web.js:22]: I think this can go now that the rest of the PRs have landed now, right?

File: react/features/base/ui/middleware.web.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        // Remove the following line only for testing purposes.
+        break;
+
+        // eslint-disable-next-line no-unreachable
+        const { language } = i18next;
+        const direction = detectDirectionBasedOnLanguage(language);
+
+        // Make sure the dir attribute is set on the body, otherwise native components will break.
+        document.getElementsByTagName('Body')[0].dir = direction;
```
Comment (saghul) [inline react/features/base/ui/middleware.web.js:31]: Capital B?

File: react/features/base/ui/actions.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Sets the direction of app; Whether it should be LTR or RTL.
+ *
+ * @param {string} direction - Direction of app.
+ * @returns {{
+ *     type: SET_DIRECTION,
+ *     direction: string,
+ * }}
+ */
+export function setDirection(direction) {
```
Comment (saghul) [inline react/features/base/ui/actions.js:13]: setLanguageDirection

**Review by saghul — CHANGES_REQUESTED** [review]
Left some comments, PTAL and rebase, thank you!

File: react/features/base/ui/middleware.web.js (hunk 5 above)
Comment (c0m1t) [reply react/features/base/ui/middleware.web.js:22]: No. There are still changes that need to be made. If you select a RTL language, and u open chat, it will break the layout. Or Filmstrips will break.

File: react/features/base/ui/components/JitsiThemeProvider.web.js (hunk 4 above)
Comment (c0m1t) [reply react/features/base/ui/components/JitsiThemeProvider.web.js]: I stored the `muiBrandedRtlTheme` state in the store, when there is a `customTheme`. But I have not tested it.

File: react/features/base/ui/reducer.web.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * @param {Object} state - The Redux state of the feature features/base/ui.
+ * @param {Object} action - Action object.
+ * @returns {Object}
+ */
+ReducerRegistry.register(
+    'features/base/ui',
+    (state = {}, action) => {
+        switch (action.type) {
+        case SET_LANGUAGE_DIRECTION: {
+
```
Comment (saghul) [inline react/features/base/ui/reducer.web.js:17]: style: unnecessary newline

**Review by saghul — COMMENTED** [review]
The chnage to the custom theme LGTM

File: react/features/base/ui/reducer.web.js (hunk 8 above)
Comment (c0m1t) [reply react/features/base/ui/reducer.web.js]: Fixed.
